experiment/selinux/src/selinux.rs

```diff
@@ -0,0 +1,302 @@
+use std::sync::Once;
+use crate::xattr::*;
+use nix::unistd::gettid;
+use nix::sys::statfs;
+use nix::errno::Errno;
+use std::path::PathBuf;
+use std::fs::File;
+use std::io::{self, Read};
+use std::os::fd::{AsFd, AsRawFd};
+
+const XATTR_NAME_SELINUX: &str = "security.selinux";
+const ERR_EMPTY_PATH: &str = "empty path";
+static ATTR_PATH_INIT: Once = Once::new();
+static mut HAVE_THREAD_SELF: bool = false;
+
+pub fn set_disabled() {
+    panic!("not implemented yet")
+}
+
+pub fn get_enabled() -> bool {
+    panic!("not implemented yet")
+}
+
+pub fn class_index(class: &str) -> Result<i64, String> {
+    panic!("not implemented yet")
+}
+
+// set_file_label sets the SELinux label for this path, following symlinks, or returns an error.
+pub fn set_file_label(fpath: &str, label: &str) -> Result<(), std::io::Error> {
+    if fpath.is_empty() {
+        return Err(std::io::Error::new(io::ErrorKind::InvalidInput, ERR_EMPTY_PATH));
+    }
+
+    loop {
+        match set_xattr(fpath, XATTR_NAME_SELINUX, label.as_bytes(), 0) {
+            Ok(_) => break,
+// TODO            Err(Errno::EINTR) => continue,
```

Comment:
Is there specific reason for keeping this TODO for a next PR?
Same for other todos below

Reply:
Thanks for your comment!
Some functions like this depend on xattr library, which is not implemented yet.
Therefore, I keep this TODO. Will fix it after implementing xattr.
(I will add this explanation in the code as well)